contrib/benitlux/src/benitlux_model.nit

```diff
@@ -85,3 +86,190 @@ class BeerEvents
 		return ""
 	end
 end
+
+# Basic user public information
+class User
+	serialize
+
+	# ID of this user in the DB
+	var id: Int
+
+	# Visible user name
+	var name: String
+
+	redef fun to_s do return "<{class_name} {id} '{name}'>"
+end
+
+# User public information in reference to a user (medium details level)
+class UserAndFollowing
+	serialize
+
+	# Basic user information
+	var user: User
+
+	# Favorite beers as a string
+	var favs: String
+
+	# Is the reference user following `user`?
+	var following: Bool
+
+	# Is the reference user followed by `user`?
+	var followed: Bool
+
+	# Time of the last check in, if `followed`
+	var last_check_in: nullable String = null
+end
+
+# Review by `author` on a `beer`
+class Review
+	serialize
+
+	# Basic author information
+	var author: User
+
+	# Basic beer information
+	var beer: Beer
+
+	# Rating out of 5
+	var rating: Int
+
+	# Text content of this review, may be empty
+	var text: String
+end
+
+# Statistics of the reviews on `beer`
+class BeerStats
+	serialize
+
+	# Basic beer information
+	var beer: Beer
+
+	# Average rating out of 5
+	var average: Float
+
+	# Number of reviews used for these statistics
+	var count: Int
+end
+
+# Beer information in reference to a user (medium detail level)
+class BeerAndRatings
+	serialize
+
+	# Basic beer information
+	var beer: Beer
+
+	# Global statistics on `beer`
+	var global: nullable BeerStats
+
+	# Statistics on `beer` among followed users only
+	var followed: nullable BeerStats
+
+	# Rating left by the referenced user, if any
+	var user_rating: nullable Int
+
+	# Do we predict this `beer` to go quickly?
+	var will_go_fast: Bool
+
+	# Data of appearance of this batch TODO
+	var batch: String
+
+	# Is this beer new since the last work day?
+	var is_new: Bool
+
+	# Has this beer been here since the last work day?
+	var is_fix: Bool
+
+	# Has this beer left the menu since the last work day?
+	var is_gone: Bool
+
+	# Noteworthy comments on this beer, often relative to a user
+	var badges: nullable Array[BeerBadge] = null is writable
+end
+
+# Noteworthy comment on a beer
+abstract class BeerBadge
+	serialize
+end
+
+# A beer is the favorite of `users`
+class FavoriteBeerBadge
+	super BeerBadge
+	serialize
+
+	# Name of the user fan of the beer
+	var users = new Array[String]
```

Comment:
This is serialized and sent to the clients. So I try to flatten this structure as much as possible.